File: Examples/c/showcase.c
/* Time Warp C Showcase
 * Demonstrates: functions, arrays, pointers, sorting, and simple IO
 */

#include <stdio.h>
#include <stdlib.h>
#include <time.h>

void fill_random(int *arr, int n) {
    for (int i = 0; i < n; ++i) arr[i] = rand() % 100 + 1;
}

void print_array(int *arr, int n) {
    for (int i = 0; i < n; ++i) printf("%3d ", arr[i]);
    printf("\n");
}

void sort_asc(int *arr, int n) {
    for (int i = 0; i < n-1; ++i) {
        for (int j = i+1; j < n; ++j) {
            if (arr[i] > arr[j]) {
                int t = arr[i]; arr[i] = arr[j]; arr[j] = t;
            }
        }
    }
}

int sum_array(int *arr, int n) {
    int s = 0;
    for (int i = 0; i < n; ++i) s += arr[i];
    return s;
}

int main(void) {
    srand((unsigned)time(NULL));
    const int N = 12;
    int nums[N];
    printf("Time Warp C Showcase\n");
    fill_random(nums, N);
    printf("Original: "); print_array(nums, N);
    sort_asc(nums, N);
    printf("Sorted:   "); print_array(nums, N);
    int total = sum_array(nums, N);
    printf("Sum = %d, Avg = %d\n", total, total / N);

    printf("Pick a threshold to display numbers >= threshold: ");
    int thr = 50;
    if (scanf("%d", &thr) == 1) {
        printf("Numbers >= %d:", thr);
        for (int i = 0; i < N; ++i) if (nums[i] >= thr) printf(" %d", nums[i]);
        printf("\n");
    } else {
        printf("No threshold entered — skipping.\n");
    }
    return 0;
}

File: Examples/pascal/showcase.pas
program ShowcasePascal;
{
  Time Warp Pascal Showcase
  Demonstrates: structured program, functions, arrays, user input, looping
}

uses SysUtils;

const
  MAX = 10;

type
  TIntArray = array[1..MAX] of Integer;

var
  nums: TIntArray;
  i, sum, avg: Integer;

function ReadInt(prompt: string): Integer;
var
  s: string;
  v: Integer;
begin
  Write(prompt);
  ReadLn(s);
  Val(s, v, i);
  ReadInt := v;
end;

procedure FillAndShow(var a: TIntArray);
var
  i: Integer;
begin
  for i := 1 to MAX do
  begin
    a[i] := Random(100) + 1;
    WriteLn('Item ', i, ': ', a[i]);
  end;
end;

function SumArray(const a: TIntArray): Integer;
var
  i, s: Integer;
begin
  s := 0;
  for i := 1 to MAX do s := s + a[i];
  SumArray := s;
end;

begin
  Randomize;
  WriteLn('Time Warp Pascal Showcase');
  FillAndShow(nums);
  sum := SumArray(nums);
  avg := sum div MAX;
  WriteLn('Sum = ', sum, ' Average = ', avg);
  i := ReadInt('Enter a threshold to report numbers >= threshold: ');
  WriteLn('Numbers >= ', i, ':');
  for i := 1 to MAX do if nums[i] >= i then WriteLn('  ', nums[i]);
  WriteLn('Done.');
end.
